tools/protobuf/IceRpc.ProtocGen/ClientGenerator.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) ZeroC, Inc.
+
+using Google.Protobuf.Reflection;
+using IceRpc.CaseConverter.Internal;
+
+namespace IceRpc.ProtocGen;
+
+internal class ClientGenerator
+{
+    internal static string GenerateInterface(ServiceDescriptor service)
+    {
+        string operations = "";
+        foreach (MethodDescriptor method in service.Methods)
+        {
+            string inputType = method.InputType.GetFullyQualifiedType();
+            string returnType = method.OutputType.GetFullyQualifiedType();
+            string methodName = $"{method.Name.ToPascalCase()}Async";
+
+            operations += $@"
+    global::System.Threading.Tasks.Task<{returnType}> {methodName}(
+        {inputType} message,
+        IceRpc.Features.IFeatureCollection? features = null,
+        global::System.Threading.CancellationToken cancellationToken = default);";
+        }
+        return @$"
+public partial interface I{service.Name.ToPascalCase()}
+{{
+    {operations.Trim()}
+}}";
+    }
+
+    internal static string GenerateImplementation(ServiceDescriptor service)
+    {
+        string operations = "";
+        foreach (MethodDescriptor method in service.Methods)
+        {
+            string inputType = method.InputType.GetFullyQualifiedType();
+            string returnType = method.OutputType.GetFullyQualifiedType();
+            string methodName = $"{method.Name.ToPascalCase()}Async";
+
+            MethodOptions? methodOptions = method.GetOptions();
+            bool idempotent =
+                (methodOptions?.HasIdempotencyLevel ?? false) &&
+                (methodOptions.IdempotencyLevel == MethodOptions.Types.IdempotencyLevel.NoSideEffects ||
+                 methodOptions.IdempotencyLevel == MethodOptions.Types.IdempotencyLevel.Idempotent);
+
+            operations += $@"
+    public global::System.Threading.Tasks.Task<{returnType}> {methodName}(
+        {inputType} message,
+        IceRpc.Features.IFeatureCollection? features = null,
+        global::System.Threading.CancellationToken cancellationToken = default) =>
+        IceRpc.Protobuf.InvokerExtensions.InvokeAsync(
```

Comment:
For IceRPC + Slice, we generate:
```csharp
using IceRpc.Slice;
```

I think we should do likewise for IceRPC + Protobuf. And then this can be a simpler `Invoker.InvokerAsync(...)`.